uitools/import/Esri/ArcGISRuntime/Toolkit/PopupView.qml

```diff
@@ -216,6 +213,20 @@ Control {
                 }
             }
         }
+
+        Button {
+            anchors {
+                top: fieldsLayout.bottom
+                topMargin: 20
```

Comment:
:x: Try not hard-coding margin/padding values into toolkit QML. Sometimes this can make tools look "off" when the developer applies a QML style on top of the toolkit item. I'm attempting to actively expunge the hard-coded values that still actively exist. 

If you are attempting to put some space between two content items, a better approach is to set `topMargin: popupView.spacing`. The property [spacing](https://doc.qt.io/qt-5/qml-qtquick-controls2-control.html#spacing-prop) is designed for this use case.

You can set the default spacing of the `PopupView` to 20 if you wish - but this at least gives the developer a meaningful chance to tweak the value if the styling doesn't apply right.

Reply:
I was not aware of that, but that totally makes sense. I'm removing it for this PR and will avoid this in the future

Reply:
@anmacdonald Removing the hardcoded value of 20 and replacing it with spacing does not seem to work and so the close button ends up awkwardly close to the text above it. Is there a different property I could try? I'm going to play around with the qml and see if I can figure it out but you may have the answer a lot quicker :) 

![spacing](https://user-images.githubusercontent.com/3746552/134411412-ec53619c-274b-41cf-bb66-a3e7fd0bda9d.PNG)

Reply:
Sorry Gela, that's my fault for not being clear.

I'll take the chance to go into a bit more detail. The default spacing of `Control`  is 0, and it's up to the inherited types like `Button`, `DialogButtonBar` etc to set sensible defaults. You can see this happen under `$QT_INSTALL/qml/QtQuick/Controls.2`

For example, with Button, we can see this happen in `$QT_INSTALL/qml/QtQuick/Controls.2/Button.qml`:

```qml
T.Button {
    id: control

    implicitWidth: Math.max(implicitBackgroundWidth + leftInset + rightInset,
                            implicitContentWidth + leftPadding + rightPadding)
    implicitHeight: Math.max(implicitBackgroundHeight + topInset + bottomInset,
                             implicitContentHeight + topPadding + bottomPadding)

    padding: 6
    horizontalPadding: padding + 2
    spacing: 6
    ...
```

In this case, `Popup` also inherits directy from `Control` (my fault - it would be better if it it inherited from `Frame`), so you also need to set a sensible `spacing` default.

So I'd put `spacing: 20` somewhere on the `PopupStackView`, and as an inherited propery it will allow the developer to change the spacing value on `PopupStackView` without having to go in and edit the source code.